statics/js/components/graph-layout.js

```diff
@@ -1,6 +1,188 @@
+      const elapsedMillis = Date.now() - new Date(metrics.Last);
+
+      if (deltaMillis === 0) {
+        return 0;
+      }
+      if (elapsedMillis > maxClockSkewMillis) {
+        return 0;
+      }
+
+      return Math.floor(8 * totalByte * 1000 / deltaMillis); // bits-per-second 
+    },
+
+    setup: function(topology) {
+      this.topology = topology;
+    },
+
+    updateData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return;
+      }
+
+      const defaultBandwidthBaseline = 1024 * 1024 * 1024; // 1 gbps
+
+      link.bandwidthBaseline = (this.topology.bandwidth.bandwidthThreshold === 'relative') ?
+        link.target.metadata.Speed || defaultBandwidthBaseline : 1;
+
+      link.bandwidthAbsolute = this.bandwidthFromMetrics(link.target.metadata.LastUpdateMetric);
+      link.bandwidth = link.bandwidthAbsolute / link.bandwidthBaseline;
+    },
+ 
+    hasData: function(link) {
+      if (!link.target.metadata.LastUpdateMetric) {
+        return false;
+      }
+
+      if (!link.bandwidth) {
+        return false;
+      }
+      return link.bandwidth > this.topology.bandwidth.active;
+    },
+
+    getText: function(link) {
+      return bandwidthToString(link.bandwidthAbsolute);
+    },
+
+    isActive: function(link) {
+      return (link.bandwidth > this.topology.bandwidth.active) && (link.bandwidth < this.topology.bandwidth.warning);
+    },
+
+    isWarning: function(link) {
+      return (link.bandwidth >= this.topology.bandwidth.warning) && (link.bandwidth < this.topology.bandwidth.alert);
+    },
+
+    isAlert: function(link) {
+      return link.bandwidth >= this.topology.bandwidth.alert;
+    },
+  },
+});
+
+var LinkLabelLatency = Vue.extend({
+  mixins: [apiMixin],
+
+  methods: {
+    setup: function(topology) {
+      this.active = 0;
+      this.warning = 10
+      this.alert = 100;
+    },
+
+    flowKey: function(a) {
+      return ${a.TrackingID};
+    },
+
+    flowCompare: function(a, b) {
+      const keya = this.flowKey(a);
+      const keyb = this.flowKey(b);
+      return keya.localeCompare(keyb);
+    },
+
+    updateLatency: function(link, flows) {
+      let prev, curr;
+      for (let i in flows) {
+        prev = curr;
+        curr = flows[i];
+
+        if (!prev) {
+          continue;
+        }
+
+        if (this.flowKey(curr) != this.flowKey(prev)) {
+          continue;
+        }
+
+        link.latencyTimestamp = Math.max(curr.Last, prev.Last);
+        link.latency = Math.abs(curr.RTT - prev.RTT) / 1000000;
+        return true;
+      }
+      return false;
+    },
+
+    updateData: function(link) {
+      var self = this;
+
+      const a = link.source.metadata;
+      const b = link.target.metadata;
+
+      if (!a.Capture) {
+        return;
+      }
+      if (!b.Capture) {
+        return;
+      }
+
+      const limit = 10;
+      const context = Date.now() - maxClockSkewMillis;
+      const query = `G.Flows()` +
```

Comment:
It won't fix the issue as using limit can remove the flow that we expect. My suggestion was to use Flow.Start since really few minutes removing limit. Like Flows().Has('Start', GT())

Reply:
I still want some limitation as there may be cases in which I had a burst of activity in the last minute(s) ... so I would want two restriction elements in place:
- limit flow count - can be 10, 100, 1000 but we need a set limit to protect the UI
- restrict to relevant time slice - you proposed `.Host('Start' ...)` I am using `Context(..)` is this reasonable (even if not equivalent)?